File: frontend/src/app/utils/employee-role.ts
import {LicenseCategory} from "./license-category";

/**
 * Role pracowników.
 */
export enum EmployeeRole {

  ADMINISTRATOR,
  DRIVING_INSTRUCTOR_A,
  DRIVING_INSTRUCTOR_B,
  LECTURER,
  DELETED,
  NONE

}

export namespace EmployeeRole {

  export function getInstructorRoleByCourseCategory(category: LicenseCategory): EmployeeRole {
    switch (category.toString()) {
      case "AM":
      case "A1":
      case "A2":
      case "A":
        return EmployeeRole.DRIVING_INSTRUCTOR_A;
      case "B":
        return EmployeeRole.DRIVING_INSTRUCTOR_B;
      default:
        return EmployeeRole.NONE;
    }
  }

}

File: src/main/java/com/project/webapp/drivingschool/data/utils/EmployeeRole.java
package com.project.webapp.drivingschool.data.utils;

/**
 * Rola pracownika szkoły nauki jazdy
 */
public enum EmployeeRole {

    /**
     * Administrator systemu
     */
    ADMINISTRATOR,

    /**
     * Instruktor jazd szkoleniowych dla wszystkich typów kategorii A (AM, A1, A2, A)
     */
    DRIVING_INSTRUCTOR_A,

    /**
     * Instruktor jazd szkoleniowych dla kategorii B
     */
    DRIVING_INSTRUCTOR_B,

    /**
     * Wykładowca, prowadzący zajęcia teoretyczne dla wszystkich kategorii
     */
    LECTURER,

    /**
     * Pracownik usunięty
     */
    DELETED;

    /**
     * Sprawdzanie, czy pracownik jest wykładowcą.
     *
     * @return true - jeśli tak, false - w przeciwnym razie
     */
    public boolean isLecturer() {
        return LECTURER.equals(this);
    }

    /**
     * Sprawdzanie, czy pracownik jest instruktorem jazdy.
     *
     * @return true - jeśli tak, false - w przeciwnym razie
     */
    public boolean isInstructor() {
        return DRIVING_INSTRUCTOR_A.equals(this) || DRIVING_INSTRUCTOR_B.equals(this);
    }

}
